### ISSUE TITLE
Invalid Version Check Output Error

### ISSUE THREAD

**Issue by dojima** [streamlink/streamlink-twitch-gui#591]
### Checklist

- [x] This is a bug report.
- [ ] This is a feature request.
- [x] I understand the difference between *Streamlink* and *Streamlink Twitch GUI*.
- [x] I tried solving my issue by reading the project's wiki first.
- [x] I have read the contribution guidelines and used the search function to find already existing threads.

### Description

'Invalid version check output' when using Streamlink Portable found [here.](https://github.com/beardypig/streamlink-portable#streamlink-portable)

### Expected / Actual behavior

Stream launches.

### Reproduction steps

1. Click any stream

### Environment details

Operating system and version: Windows 7 x64
Streamlink Twitch GUI version: 1.5.0
Streamlink version: 0.14.2

### Comments, logs, screenshots, etc.

As an aside, I always dread the moment it becomes necessary to update Streamlink.  I try to put it off as long as possible by only updating the plugins because I know that each time I do update Streamlink itself, something is going to go wrong.  Unfortunately, this time I can't fix it myself.  I've been using Streamlink since it was Livestreamer, and I remember Livestreamer just being a single, self-contained (more or less) executable that simply worked.  It didn't need a batch file, I didn't need to mess around with Python scripts, it didn't need a folder with a Python executable, and it didn't need to set PATH directories.

I don't really know enough (read: anything) about programming to make a conclusion as to whether or not the way Streamlink currently works in relation to Livestreamer is required or better, all I can say for certain is that it's far more problematic and less flexible from the standpoint of an end user.  That being said, this issue is about Streamlink Twitch GUI and not Streamlink, so I'll cut the rant short there.

Can I just do something simple to ignore version checking?  Everything works perfectly well when launching streams from command-line.

Thanks.

**Comment by bastimeyer** [comment]
I haven't used beardypig's Streamlink portable yet, but similar to the other portable Streamlink version, you have to set the paths of both `pythonw.exe` and `streamlink-script.py` / `Streamlink.py` in the Twitch GUI's streaming settings, as both portable versions are packaged differently and require custom settings so that it can be validated.

The "invalid version check output" error is a bit awkward and needs to be improved, as it isn't clear to the average user what it actually means. It means that while checking Streamlink's version (which is required for knowing its features) something went wrong while executing it and the validation failed due to invalid output. Returning a wrong version would yield a different error message. I will have to rewrite and improve this in the future when I'm going to rewrite parts of the streaming dialog.

If you're having problems setting up and configuring Streamlink and Streamlink Twitch GUI, just follow the wiki pages. It really isn't that difficult.
https://github.com/streamlink/streamlink-twitch-gui/wiki/Installation
https://github.com/streamlink/streamlink-twitch-gui/wiki/Streamlink-configuration

Livestreamer was packaged differently and was released as a standalone executable. Streamlink doesn't do that when you're using the Windows portable versions or the Windows installer and you'll get a local python environment (the executables), the Streamlink python modules together with all of their dependencies and also the precompiled python modules built for that specific python environment. However, if you're using the Windows installer, Streamlink Twitch GUI should be able to automatically resolve both Streamlink as well as Python and should validate it without having to set any custom settings.

**Comment by MrLint** [comment]
While python is kinda outside of my wheelhouse, just as a comparison of the  RosadinTV in both versions
Streamlink.exe -V gives : 

[Streamlink for Windows v0.11.0 - Git 64c22e0]
Streamlink.exe 0.11.0
[End of Streamlink for Windows with ExitCode 0]

vs

[Streamlink for Windows v0.14.0 - Git 4f5bb1e]
Streamlink.exe 0.14.0@4f5bb1e
[End of Streamlink for Windows with ExitCode 0]

My first reaction would be that it does not like the @ in the string output.

**Comment by dojima** [comment]
Yeah, I have all the paths set up.  I've always been using the portable version since moving to Streamlink, and I've always been able to make it work until now.  I've also implemented the shebang explicit path thing mentioned [here](https://github.com/streamlink/streamlink-twitch-gui/issues/451#issuecomment-302595543), if that's relevant.

I suppose the issue could be similar to [this](https://github.com/streamlink/streamlink-twitch-gui/issues/581). I could try opening an issue on beardypig's portable Streamlink project if you think that's advisable.

Mid-post edit:  Yeah, MrLint, I was thinking something like that.  If I do streamlink -V here, my output is

`streamlink 0+unknown`

Thanks, all.

**Comment by bastimeyer** [comment]
@MrLint 
That's caused by an old Streamlink Portable version, which got fixed (see streamlink/streamlink-portable#26)

@dojima 
As I've said, I haven't used @beardypig's Streamlink portable yet, but if it's returning a 0 as version (the `+unknown` can be ignored), it's definitely a bug which needs to get fixed over there. This may be caused by a change in Streamlink recently where the version gets retrieved dynamically. I've tagged him, so maybe he can check this.

**Comment by bastimeyer** [comment]
I should have noticed this sooner. You were talking about the "**invalid** version check output" instead of the "**unexpected** version check output", so of course it had to be caused by Streamlink portable.

- https://github.com/streamlink/streamlink-twitch-gui/blob/v1.6.0/src/app/services/streaming/provider/validate.js#L53
- https://github.com/streamlink/streamlink-twitch-gui/blob/v1.6.0/src/app/services/streaming/provider/validate.js#L62

**Comment by MrLint** [comment]
@bastimeyer Yep i was looking thru the code and that is the same place I thought it'd be.

**Comment by beardypig** [comment]
@dojima feel free to open an issue on my streamlink-portable repo. The reason a bat file is used is to ensure that streamlink can find the portable `ffmpeg`, `rtmpdump`, and `streamlinkrc` file (but these are not likely required when streaming twitch). I cannot reproduce the version return `0+unknown` issues though, I get `streamlink portable-0.14.2-py3.6.5-win32` - which I didn't actually expect...

**Comment by beardypig** [comment]
@dojima this should be fixed in all future versions (stable and nightly), and I have fixed it for the current version [0.14.2+1](https://github.com/beardypig/streamlink-portable/releases/tag/0.14.2%2B1).

**Comment by MrLint** [comment]
@bastimeyer I was reading the other thing you linked to, So short of that build chaining its token, or having some added string checking in this project its just gonna be how it is for the RosadinTV build?

**Comment by bastimeyer** [comment]
Just download the latest Streamlink portable from RosadinTV again and rebuild Streamlink. The version string will be fixed then.

I'm not going to change anything here regarding version strings which are not [SemVer](https://semver.org/) compliant.

**Comment by MrLint** [comment]
@bastimeyer ok I'll double check this when I get home. I downloaded v14 last night from the link on the SL portable page. I'll dig further in to make sure its just not a stale link.
